azure-kusto-data/src/connection_string.rs

```diff
@@ -364,17 +376,32 @@ impl ConnectionStringAuth {
                 client_authority,
             } => Arc::new(ClientSecretCredential::new(
                 azure_core::new_http_client(),
+                Url::parse(client_authority.as_str()).unwrap(),
                 client_authority,
                 client_id,
                 client_secret,
-                TokenCredentialOptions::default(),
             )),
             ConnectionStringAuth::ApplicationCertificate { .. } => unimplemented!(),
-            ConnectionStringAuth::ManagedIdentity { user_id } => {
+            ConnectionStringAuth::ManagedIdentity { user_id ,
+                client_id,
+                tenant_id,
+                client_authority,} => {
+                    
+                let authority_host: Url = Url::parse("https://login.microsoftonline.com").unwrap();
                 if let Some(user_id) = user_id {
-                    Arc::new(ImdsManagedIdentityCredential::default().with_object_id(user_id))
+                    Arc::new(WorkloadIdentityCredential::new(             
```

Comment:
As far as I know, this is not equivalent.
Workload Identities are a k8s thing, and Managed Identities are a more general azure thing.

If I'm incorrect, please provide a source.

The PR we need was released in 0.23.0, and right now we still can't upgrade to that version due to the other deps.